spec/javascripts/accordion-with-descriptions-spec.js

```diff
@@ -2,215 +2,242 @@ describe('An accordion with descriptions module', function () {
   "use strict";
 
   var $element;
-
-  beforeEach(function() {
-    $element = $('<div class="subsections js-hidden" data-module="accordion-with-descriptions">\
-        <div class="subsection-wrapper">\
-          <div class="subsection">\
-            <div class="subsection__header">\
-              <h2 class="subsection__title">Subsection title in here</h2>\
-              <p class="subsection__description">Subsection description in here</p>\
-            </div>\
-            <div class="subsection__content" id="subsection_content_0">\
-              <ul class="subsection__list">\
-                <li class="subsection__list-item">\
-                  <a href="">Subsection list item in here</a>\
-                </li>\
-              </ul>\
-            </div>\
+  var html = '\
+    <div class="subsections js-hidden" data-module="accordion-with-descriptions">\
+      <div class="subsection-wrapper">\
+        <div class="subsection">\
+          <div class="subsection__header">\
+            <h2 class="subsection__title" id="topic-section-one">Topic Section One</h2>\
+            <p class="subsection__description">Subsection description in here</p>\
           </div>\
-          <div class="subsection">\
-            <div class="subsection__header">\
-              <h2 class="subsection__title">Subsection title in here</h2>\
-              <p class="subsection__description">Subsection description in here</p>\
-            </div>\
-            <div class="subsection__content" id="subsection_content_1">\
-              <ul class="subsection__list">\
-                <li class="subsection__list-item">\
-                  <a href="">Subsection list item in here</a>\
-                </li>\
-              </ul>\
-            </div>\
+          <div class="subsection__content" id="subsection_content_0">\
+            <ul class="subsection__list">\
+              <li class="subsection__list-item">\
+                <a href="">Subsection list item in here</a>\
+              </li>\
+            </ul>\
           </div>\
         </div>\
-      </div>');
-
-    var accordion = new GOVUK.Modules.AccordionWithDescriptions();
-    accordion.start($element);
-  });
+        <div class="subsection">\
+          <div class="subsection__header">\
+            <h2 class="subsection__title" id="topic-section-two">Topic Section Two</h2>\
+            <p class="subsection__description">Subsection description in here</p>\
+          </div>\
+          <div class="subsection__content" id="subsection_content_1">\
+            <ul class="subsection__list">\
+              <li class="subsection__list-item">\
+                <a href="">Subsection list item in here</a>\
+              </li>\
+            </ul>\
+          </div>\
+        </div>\
+      </div>\
+    </div>';
 
   afterEach(function() {
     $(document).off();
   });
 
-  // Setup
+  describe('when no sub section is linked to', function () {
```

Comment:
It's to allow for two different `beforeEach` behaviours. I need to mock out `getCurrentLocation` before `start` is called, but only in the specific tests. If you have a 'global' multiple `beforeEach` and a nested one they _both_ get called and then you have two lots of setup (the first is just never used) and that feels weird and is hard to debug…

One alternative I guess would be to get rid of the `beforeEach` setup and do it in every single test individually?

Open to suggestions.